src/reader/data_loader_reader.py

```diff
@@ -0,0 +1,93 @@
+"""
+ Copyright (C) 2020  Argonne, Hariharan Devarajan <hdevarajan@anl.gov>
+ This file is part of DLProfile
+ DLIO is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as
+ published by the Free Software Foundation, either version 3 of the published by the Free Software Foundation, either
+ version 3 of the License, or (at your option) any later version.
+ This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ details.
+ You should have received a copy of the GNU General Public License along with this program.
+ If not, see <http://www.gnu.org/licenses/>.
+"""
+import math
+import logging
+import numpy as np
+
+from src.utils.utility import utcnow
+
+from torch.utils.data import Dataset, DataLoader
+from torch.utils.data.distributed import DistributedSampler
+
+from src.common.enumerations import Shuffle
+from src.reader.reader_handler import FormatReader
+
+class DataLoaderReader(FormatReader):
+    """
+    DataLoader reader and iterator logic.
+    PyTorch data loader is file format agnostic so it is not technically separate from the other formats
+    For now we will assume we are reading .npz files through the dataloader
+    """
+    class NpzDataset(Dataset):
```

Comment:
This class can move in DataReader. We can have a factory to load the correct data reader generically.

Reply:
What would this look like, considering we already have npz_reader which is not the same as this? For all data formats we want to support with PT dataloader, we might need similar Dataset classes. See comment the README for more details.

Reply:
I was thinking in these lines.

we can have a base reader which is actual calls to npz.
If we want to change how its implemented on pytorch or tf, we can extend it and do the extensions appropriately.
This way we have a cleaner abstraction. What do you think?